src/cluster.c

```diff
@@ -5959,7 +5962,13 @@ clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, in
     /* If we don't have all the keys and we are migrating the slot, send
      * an ASK redirection. */
     if (migrating_slot && missing_keys) {
-        if (error_code) *error_code = CLUSTER_REDIR_ASK;
+        if (error_code) {
+            if (existing_keys) {
+                *error_code = CLUSTER_REDIR_UNSTABLE;
```

Comment:
A minor point. setting the error code to unstable should probably go with returning NULL, we aren't directing in this case.

Reply:
updated.